File: TeachMeWeb/src/app/user/tutor.ts
import {Course} from "../course/course";

export class Tutor {

  private _idTutor: number;
  private _username: string;
  private _password: string;
  private _mail: string;
  private _phone: string;
  private _evaluation: number;
  private _description: string;
  private _isWarned: boolean;
  private _isModerator: boolean;
  private _year:number;
  private _section:string;
  private _token:string;



  constructor(username: string="tutor", password: string="password", mail: string="tutor@hotmail.com",
              tel: string="0987654321", evaluation: number=0, description: string="", isWarned: boolean=false, isModerator: boolean=false
  ,year: number = 1,section:string="undifined",  token: string ="undefined") {
    this._username = username;
    this._password = password;
    this._mail = mail;
    this._phone = tel;
    this._evaluation = evaluation;
    this._description = description;
    this._isWarned = isWarned;
    this._isModerator = isModerator;
    this._year=year;
    this._section=section;
    this._token=token;
  }

  public deserializable(json: any) : Tutor {
    Object.assign(this, json);
    return this;
  }

  public serialize() : any {
    return {
      // idTutor: this._idTutor,
      username: this._username,
      password: this._password,
      mail: this._mail,
      phone: this._phone,
      //  evaluation: this._evaluation,
      description: this._description
      /*   isWarned: this._isWarned,
         isModerator: this._isModerator,
         year: this._year,
         section:this._section*/
    };
  }

  public toJson() : any {
    return {
      idTutor: this._idTutor,
      username: this._username,
      password: this._password,
      mail: this._mail,
      phone: this._phone,
      description: this._description,
      evaluation: this._evaluation,
      isWarned: this._isWarned,
      token: this._token,
      isModerator: this._isModerator
    }
  }

  public serializeUpdate() : any {
    return {
      idTutor: this._idTutor,
      username: this._username,
      password: this._password,
      mail: this._mail,
      phone: this._phone,
      //  evaluation: this._evaluation,
      description: this._description,
      token: this._token
    };
  }


  get year(): number {
    return this._year;
  }

  get section(): string {
    return this._section;
  }

  get idTutor(): number {
    return this._idTutor;
  }

  set idTutor(value: number) {
    this._idTutor = value;
  }

  get username(): string {
    return this._username;
  }

  set username(value: string) {
    this._username = value;
  }

  get password(): string {
    return this._password;
  }

  set password(value: string) {
    this._password = value;
  }

  get mail(): string {
    return this._mail;
  }

  set mail(value: string) {
    this._mail = value;
  }

  get phone(): string {
    return this._phone;
  }

  set phone(value: string) {
    this._phone = value;
  }

  get evaluation(): number {
    return this._evaluation;
  }

  set evaluation(value: number) {
    this._evaluation = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get isWarned(): boolean {
    return this._isWarned;
  }

  set isWarned(value: boolean) {
    this._isWarned = value;
  }

  get isModerator(): boolean {
    return this._isModerator;
  }

  set isModerator(value: boolean) {
    this._isModerator = value;
  }

  set year(value: number) {
    this._year = value;
  }

  set section(value: string) {
    this._section = value;
  }

  get token(): string {
    return this._token;
  }

  set token(value: string) {
    this._token = value;
  }
}

File: TeachMeAPI/TeachMeAPI/Models/Tutor.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Web;
using TeachMeAPI.DAO;

namespace TeachMeAPI.Models
{
    public class Tutor
    {
        [JsonProperty(PropertyName = "idTutor")]
        public int IdTutor { get; set; }
        [JsonProperty(PropertyName = "username")]
        public string UserName { get; set; }
        [JsonProperty(PropertyName = "mail")]
        public string Mail { get; set; }
        [JsonProperty(PropertyName = "password")]
        public string Password { get; set; }
        [JsonProperty(PropertyName = "isWarned")]
        public bool IsWarned { get; set; }
        [JsonProperty(PropertyName = "phone")]
        public string Phone { get; set; }
        [JsonProperty(PropertyName = "isModerator")]
        public bool IsModerator { get; set; }
        [JsonProperty(PropertyName = "description")]
        public string Description { get; set; }
        [JsonProperty(PropertyName = "evaluation")]
        public double Evaluation { get; set; }
        [JsonProperty(PropertyName = "token")]
        public string Token { get; set; }

        public Tutor()
        { }

        public Tutor(int id, string name, string mail, string password, string tel, bool avertissement, bool isModerateur, string description, double evaluation, string token)
        {
            IdTutor = id;
            UserName = name;
            Mail = mail;
            Password = password;
            Phone = tel;
            IsWarned = avertissement;
            IsModerator = isModerateur;
            Description = description;
            Evaluation = evaluation;
            Token = token;
        }

        public Tutor(SqlDataReader reader)
        {
            IdTutor = Int32.Parse(reader[TutorDAO.COLUMN_ID].ToString());
            UserName = reader[TutorDAO.COLUMN_NAME].ToString();
            Mail = reader[TutorDAO.COLUMN_MAIL].ToString();
            Password = reader[TutorDAO.COLUMN_PASSWORD].ToString();
            IsWarned = Boolean.Parse(reader[TutorDAO.COLUMN_IS_WARNED].ToString());
            Phone = reader[TutorDAO.COLUMN_PHONE].ToString();
            IsModerator = Boolean.Parse(reader[TutorDAO.COLUMN_IS_MODERATOR].ToString());
            Description = reader[TutorDAO.COLUMN_DESCRIPTION].ToString();
            Evaluation = Double.Parse(reader[TutorDAO.COLUMN_EVALUATION].ToString());
            Token = reader[TutorDAO.COLUMN_TOKEN].ToString();

        }
    }
}